Bug 1902003: Clarification of Jobs completions column data when sorting

sg00dwin opened the pull request:
https://bugzilla.redhat.com/show_bug.cgi?id=1902003 was opened regarding “Incorrect” sorting of the Job Completions column. The column data is sorting correctly on desired `{completions}` but the content is presented as `{succeeded} of {completions}` (eg 1 of 1) .   When sorted descending order, the following visually looks incorrect listed as, 2 of 20, 3 of 8, 1 of 1. 
To alleviate this confusion I suggest we include stack the `Desired {completions}` above `Succeeded {job.status.succeeded || 0}`.

<img width="922" alt="Screen Shot 2021-01-06 at 3 36 41 PM" src="https://user-images.githubusercontent.com/1874151/103833379-e1268000-504e-11eb-900a-36f5db5731db.png">
<img width="920" alt="Screen Shot 2021-01-06 at 3 32 54 PM" src="https://user-images.githubusercontent.com/1874151/103833383-e4ba0700-504e-11eb-8b44-591e105082f3.png">

rebeccaalpert commented on frontend/public/components/job.tsx:
```diff
@@ -232,6 +194,51 @@ const JobsList: React.FC = (props) => {
+            title="pods"
+          >
+            <div>
+              {t('workload~Desired')} {completions}
```
Yeah, it's better for the translators to have the context. Thanks @rhamilto.

sg00dwin replied:
updated

rebeccaalpert reviewed (approved):
Thanks for making that change.

rhamilto commented on frontend/public/components/job.tsx:
```diff
@@ -232,6 +194,47 @@ const JobsList: React.FC = (props) => {
+            title="pods"
+          >
+            <div>{t('workload~Desired {{completions}}', { completions })}</div>
+            <div>{t('workload~Succeeded {{count}}', { count: job.status.succeeded || 0 })}</div>
```
My bad.  `count` is a reserved keyword for pluralization.  Use `succeeded` as the key instead.

spadgett commented:
If we're going to do this, would it be better to simply make them separate columns? Then you could sort either by successful or total.

sg00dwin commented:
> If we're going to do this, would it be better to simply make them separate columns? Then you could sort either by successful or total.

I assume it would depend on how valuable it is to be able to sort by most currently `succeeded` is vs the additional space constraints. Personally I think the stacking of the two works well from a content presentation scanning perspective.

<img width="1256" alt="Screen Shot 2021-01-11 at 10 53 28 AM" src="https://user-images.githubusercontent.com/1874151/104216132-a97d5680-5407-11eb-8dd8-ed643c134322.png">

spadgett commented:
cc @beanh66 for UX input

cshinn commented:
Does every column need to be sortable? The initial format is concise and pretty clear and takes less space than separate columns would

sg00dwin commented:
> Does every column need to be sortable? The initial format is concise and pretty clear and takes less space than separate columns would

Columns don't necessarily need to be sortable, though it seems like a reasonable data to sort by.

spadgett commented:
I was wondering if we shouldn't just change the sort to use the first number (succeeded) instead of desired

beanh66 commented:
That was my initial thought as well @spadgett. The column header even makes me think that's what is being displayed anyways since _Completed_ may imply "successfully completed." 

For what it's worth, for Pods we have a column called "Ready" and then we have x of y and we seem to sort on the first number. Machine sets on the other hand seem to have the same problem as jobs, where we sorting on the second number and maybe should change to sort on the first.

sg00dwin commented:
I have updated this pr to sort by `Succeeded Completions` and reverted to display the original "x of y".
<img width="1131" alt="Screen Shot 2021-01-21 at 10 36 50 AM" src="https://user-images.githubusercontent.com/1874151/105389694-22d62f80-5be6-11eb-880b-2fd38ee86b96.png">

spadgett reviewed (commented):
@sg00dwin Thanks, can you look at updating the MachineSet sort as well?

sg00dwin commented on frontend/public/components/factory/table.tsx:
```diff
@@ -131,7 +131,7 @@ const sorts = {
   ingressValidHosts,
   serviceCatalogStatus,
-  jobCompletions: (job) => getJobTypeAndCompletions(job).completions,
+  jobCompletionsSucceeded: (job) => _.toInteger(_.get(job, 'status.succeeded')),
```
updated

sg00dwin commented on frontend/public/components/job.tsx:
```diff
@@ -232,6 +194,46 @@ const JobsList: React.FC = (props) => {
       props: { className: tableColumnClasses[5] },
     },
   ];
+  const JobTableRow: RowFunction<JobKind> = ({ obj: job, index, key, style }) => {
```
This was fixed after rebasing.

sg00dwin commented:
/retest

sg00dwin commented:
/retest

spadgett commented:
/override ci/prow/ceph-storage-plugin

The ceph-storage-plugin job should only have run on PRs with changes to `frontend/packages/ceph-storage-plugin`. It failed due to an unrelated CI config change. Overriding status on this PR.

sg00dwin commented:
/retest

sg00dwin commented:
@spadgett pr updated to include sorting MachineSet Machines by ready vs desired

sg00dwin commented:
/retest

sg00dwin commented:
/retest

spadgett reviewed (approved):
/lgtm
